extensions/src/org/pentaho/platform/osgi/KarafPropertyImporter.java

```diff
@@ -0,0 +1,120 @@
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright 2015 Pentaho Corporation. All rights reserved.
+ */
+
+package org.pentaho.platform.osgi;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+import org.apache.commons.lang.StringUtils;
+
+public class KarafPropertyImporter {
+  final static String delimiter = ",";
+  protected static KarafInstance karafInstance;
+  private String importFilePath;
+  int lineNumber;
+
+  public KarafPropertyImporter( String importFilePath ) {
+    this.importFilePath = importFilePath;
+    if ( karafInstance == null ) {
+      karafInstance = KarafInstance.getInstance();
+    }
+  }
+  
+  public void process() {
+
+    BufferedReader br = null;
+    String line = "";
+    String[] fields;
+    lineNumber = 0;
+
+    try {
+
+      br = new BufferedReader( new FileReader( importFilePath ) );
+      while ( ( line = br.readLine() ) != null ) {
+        try {
+          lineNumber++;
+          if ( !line.startsWith( "#" ) && line.trim().length() > 1 ) {
+            fields = parseLine( line );
+            if ( fields[0].toLowerCase().equals( "port" ) ) {
+              if ( fields.length < 5 )
+                throw new IllegalStateException( getErrorPrefix() + "Port definition lines must contain 6 fields" );
+              karafInstance.registerProperty( new KarafInstancePort( fields[1], fields[2], fields[3], Integer
+                  .valueOf( fields[4] ), Integer.valueOf( fields[5] ) ) );
+            }
+            if ( fields[0].toLowerCase().equals( "property" ) ) {
+              if ( fields.length < 5 )
+                throw new IllegalStateException( getErrorPrefix()
+                    + "Port definition lines must contain at least 4 fields" );
+              String value = "";
+              if ( fields.length >= 4 && !StringUtils.isEmpty( fields[3] ) ) {
+                value = fields[4];
+              }
+              karafInstance.registerProperty( new KarafInstanceProperty( fields[1], fields[2], fields[3], value ) );
+
+            }
+          }
+        } catch ( Exception e ) {
+          throw new IllegalStateException( getErrorPrefix() );
+        }
+      }
+
+    } catch ( FileNotFoundException e ) {
+      e.printStackTrace();
```

Comment:
printStackTrace needs to be replaced with logging. Shouldn't we throw Exceptions here?